bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/parser/ParsingException.java

```diff
@@ -67,10 +67,8 @@ public String getMessage() {
         StackTraceElement[] st = new StackTraceElement[size];
         for (int n = 0; n < exceptions.size(); n++) {
             StackTraceElement[] ste = exceptions.get(n).getStackTrace();
-            for (int i = 0; i < ste.length; i++) {
-                st[index] = ste[i];
-                index++;
-            }
+            System.arraycopy(ste, 0, st, index, ste.length);
+            index += ste.length;
```

Comment:
Edit: Please add a test case for this method prior to refactoring.

This fancy array engineering may with some even more fancy stream magic be refactored to:
```
List<StackTraceElement> stes = exceptions.stream()//
    .map(exception -> exception.getStackTrace())//
    .flatMap(array -> Arrays.stream(array))//
    .collect(Collectors.toList());

return stes.toArray(new StackTraceElement[0]);
```

or the old list fashioned way:

```
List<StackTraceElement> ste = new ArrayList<>();
for (ParsingNestedException e : exceptions) {
    ste.addAll(Arrays.asList(e.getStackTrace()));
}

return stes.toArray(new StackTraceElement[0]);
```